build/tools/Automation/DotNet/DotNetAutomator.cs

```diff
@@ -58,8 +58,22 @@ public async Task RunAsync()
                 this.oryxSdkStorageBaseUrl = Constants.OryxSdkStorageBaseUrl;
             }
 
-            this.oryxDotNetSdkVersions = await this.httpService.GetOryxSdkVersionsAsync(
-                this.oryxSdkStorageBaseUrl + DotNetConstants.DotNetSuffixUrl);
+            string sdkVersionsUrl = this.oryxSdkStorageBaseUrl + DotNetConstants.DotNetSuffixUrl;
+
+            // A SAS token is required for the staging account.
+            if (this.oryxSdkStorageBaseUrl == Constants.OryxSdkStagingStorageBaseUrl)
+            {
+                string sasToken = Environment.GetEnvironmentVariable(Constants.OryxSdkStagingPrivateSasTokenEnvVar);
```

Comment:
is there any chance we need to run the process locally? If we run the process locally it won't have the secrets from the Oryx-pipeline-library (I am assuming the secrets are retrieved from the oryx-pipeline-library).

Reply:
I only run locally to test this. This primarily runs on GitHub Actions. If I need to run locally, I can store the token manually.